tests/notificationprofile/test_views.py

```diff
@@ -74,29 +92,213 @@ def test_notification_profile_can_update_timeslot_without_changing_pk(self):
+            settings={"phone_number": "+4747474747"},
+        )
+
+    def teardown(self):
+        connect_signals()
+
     def test_can_get_all_destinations(self):
-        response = self.user1_rest_client.get(path="/api/v2/notificationprofiles/destinations/")
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 3)
+        response = self.user1_rest_client.get(path=self.ENDPOINT)
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(len(response.data), 3, response.data)
         response_settings = [destination["settings"] for destination in response.data]
         self.assertTrue(self.synced_email_destination.settings in response_settings)
         self.assertTrue(self.non_synced_email_destination.settings in response_settings)
         self.assertTrue(self.sms_destination.settings in response_settings)
 
     def test_can_get_destination(self):
         response = self.user1_rest_client.get(
-            path=f"/api/v2/notificationprofiles/destinations/{self.synced_email_destination.pk}/"
+            path=f"{self.ENDPOINT}{self.synced_email_destination.pk}/"
         )
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
         self.assertEqual(response.data["settings"], self.synced_email_destination.settings)
 
+
+@tag("API", "integration")
+class EmailDestinationViewTests(APITestCase):
+    ENDPOINT = "/api/v2/notificationprofiles/destinations/"
+
+    def setUp(self):
+        disconnect_signals()
+        self.user1 = PersonUserFactory()
+
+        self.user1_rest_client = APIClient()
+        self.user1_rest_client.force_authenticate(user=self.user1)
+
+        timeslot1 = TimeslotFactory(user=self.user1, name="Never")
+
+        self.notification_profile1 = NotificationProfileFactory(user=self.user1, timeslot=timeslot1)
+        # Default email destination is automatically created with user
+        self.synced_email_destination = self.user1.destinations.get()
+        self.non_synced_email_destination = DestinationConfigFactory(
+            user=self.user1,
+            media=Media.objects.get(slug="email"),
+            settings={"email_address": "test@example.com", "synced": False},
+        )
+        self.notification_profile1.destinations.set([self.synced_email_destination])
+
+    def teardown(self):
+        connect_signals()
+
+    def test_can_delete_unsynced_unconnected_email_destination(self):
+        response = self.user1_rest_client.delete(
+            path=f"{self.ENDPOINT}{self.non_synced_email_destination.pk}/"
+        )
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT, response.data)
+        self.assertFalse(DestinationConfig.objects.filter(id=self.non_synced_email_destination.pk).exists())
+
+    def test_cannot_delete_synced_email_destination(self):
```

Comment:
While I mostly can grok the object of a test when it uses terms like `can_do_this_or_that`, but I find myself becoming more uncertain about the true meaning when the concept flips to `cannot`.

This is why I prefer the verb "should" in test names.  "so and so should return this or that" or "this thing should not do foo when bar is present".

```suggestion
    def test_should_raise_400_bad_request_when_asked_to_delete_synced_email_destination(self):
```

or 

```suggestion
    def test_should_not_allow_deletion_of_synced_email_destination(self):
```